starfyre/parser.py

```diff
@@ -150,25 +156,41 @@ def handle_endtag(self, tag):
         # if the tag is not found in the generic tags and custom components
         if tag not in self.generic_tags and tag not in self.components:             
             raise UnknownTagError(f'Unknown tag: "{tag}". Please review line {self.lineno} in your "{self.component_name}" component in the pyml code.')
-
-        # we need to check if the tag is a default component or a custom component
-        # if it is a custom component, we get the element from the custom components dict          
-
-        if tag not in self.generic_tags and tag in self.components:
-            component = self.components[tag]
-            tag = component.tag
         
         endtag_node = self.stack.pop()  
         self.current_depth -= 1
+
         if endtag_node.tag != "style" and endtag_node.tag != "script":
             if len(self.stack) > 0:
                 parent_node = self.stack[-1]      #this is last item/"top element" of stack
-                parent_node.children.append(endtag_node)
+
+                if endtag_node.original_name != endtag_node.tag: #this node is for "custom" component
+                    #process endtag_node children                    
+                    new_children = []
+                    is_slot_used = False
+                    for child_component in endtag_node.children:
+                        # if child_component.tag == "slot":  #TODO: USE is_slot_component INSTEAD OF "slot" AS STRING
+                        if child_component.is_slot_component:
+                            new_children.extend(self.current_children)
```

Comment:
shouldn't it be the other way round ?

`self.children.extend(new_children)`?

Reply:
> shouldn't it be the other way round ?
> 
> `self.children.extend(new_children)`?

No. `new_children` is a copy of `endtag_node.children` with `<slot>` replaced by `self.current_children`.